pkg/liqoctl/offload/namespace.go

```diff
@@ -0,0 +1,134 @@
+	logsutils "github.com/liqotech/liqo/pkg/utils/logs"
+)
+
+func HandleOffloadCommand(ctx context.Context, command *cobra.Command, args []string) error {
+	if !klog.V(4).Enabled() {
+		klog.SetLogFilter(logsutils.LogFilter{})
+	}
+
+	config, err := common.GetLiqoctlRestConf()
+	if err != nil {
+		return err
+	}
+
+	k8sClient, err := client.New(config, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	var whitelistClusterLabels argsutils.StringMap
+	if err := whitelistClusterLabels.Set(command.Flag(AcceptedLabelsFlag).Value.String()); err != nil {
+		return err
+	}
+	var blacklistClusterLabels argsutils.StringMap
+	if err := blacklistClusterLabels.Set(command.Flag(DeniedLabelsFlag).Value.String()); err != nil {
+		return err
+	}
+
+	nsOffloading := forgeNamespaceOffloading(command, args, whitelistClusterLabels, blacklistClusterLabels)
+
+	_, err = controllerutil.CreateOrUpdate(ctx, k8sClient, nsOffloading, func() error {
+		nsOffloading.Spec.PodOffloadingStrategy = forgePodOffloadingStrategy(command)
+		nsOffloading.Spec.NamespaceMappingStrategy = forgeNamespaceMappingStrategy(command)
+		nsOffloading.Spec.ClusterSelector = forgeClusterSelector(whitelistClusterLabels, blacklistClusterLabels)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func forgeNamespaceOffloading(command *cobra.Command, args []string, labels, clusterLabels argsutils.StringMap) *offloadingv1alpha1.NamespaceOffloading {
+	return &offloadingv1alpha1.NamespaceOffloading{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      consts.DefaultNamespaceOffloadingName,
+			Namespace: args[1],
+		},
+		Spec: offloadingv1alpha1.NamespaceOffloadingSpec{
+			NamespaceMappingStrategy: forgeNamespaceMappingStrategy(command),
+			PodOffloadingStrategy:    forgePodOffloadingStrategy(command),
+			ClusterSelector:          forgeClusterSelector(labels, clusterLabels),
+		},
+	}
+}
+
+func forgeClusterSelector(labels, clusterLabels argsutils.StringMap) corev1.NodeSelector {
+	var l []corev1.NodeSelectorTerm
+
+	// No labels are defined, backing to the default behavior
+	if len(labels.StringMap) == 0 && len(clusterLabels.StringMap) == 0 {
+		return corev1.NodeSelector{
+			NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+				MatchExpressions: []corev1.NodeSelectorRequirement{{
+					Key:      consts.TypeLabel,
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{consts.TypeNode},
+				}}},
+			},
+		}
+	}
+
+	for k, v := range labels.StringMap {
+		l = append(l, corev1.NodeSelectorTerm{
+			MatchExpressions: []corev1.NodeSelectorRequirement{{
+				Key:      k,
+				Operator: corev1.NodeSelectorOpIn,
+				Values:   []string{v},
+			},
+			},
+		})
+	}
+
+	for k, v := range clusterLabels.StringMap {
+		l = append(l, corev1.NodeSelectorTerm{
+			MatchExpressions: []corev1.NodeSelectorRequirement{{
+				Key:      k,
+				Operator: corev1.NodeSelectorOpNotIn,
+				Values:   []string{v},
+			},
+			},
+		})
+	}
```

Comment:
Using a different NodeSelectorTerm for each label specified limits the power of the logic; if I want a namespace replica on all clusters that have both provider = A, and region = B, I couldn't get this result with a solution like this. However, I understand that specifying if labels must be considered in "AND"/"OR" from the command line would perhaps complicate things too much. Overall, it seems like a good compromise to only allow the "OR" of labels with liqoctl.

Reply:
Yes, the idea is to make the command very easy to understand sacrificing a bit the expressiveness.